kinode/packages/app_store/app_store/src/state.rs

```diff
@@ -0,0 +1,566 @@
+                    .map_err(|_| AppStoreLogError::DecodeLogError)?;
+                let metadata_url = upd.metadataUrl;
+                let metadata_hash = upd.metadataHash;
+
+                let package_hash = log.topics()[1].to_string();
+                let metadata_hash = metadata_hash.to_string();
+
+                let Some(current_listing) =
+                    self.get_listing_with_hash_mut(&package_hash.to_string())
+                else {
+                    // package not found, so we can't update it
+                    // this will never happen if we're ingesting logs in order
+                    return Ok(());
+                };
+
+                let metadata = if update_listings {
+                    Some(utils::fetch_metadata_from_url(
+                        &metadata_url,
+                        &metadata_hash,
+                        5,
+                    )?)
+                } else {
+                    None
+                };
+
+                current_listing.metadata_url = metadata_url;
+                current_listing.metadata_hash = metadata_hash;
+                current_listing.metadata = metadata;
+
+                // if we have this app installed, and we have auto_update set to true,
+                // we should try to download new version from the mirrored_from node
+                // and install it if successful.
+                // let package_id = PackageId::new(&current_listing.name, &current_listing.publisher);
+                // if let Some(package_state) = self.downloaded_packages.get(&package_id) {
+                //     if package_state.auto_update {
+                //         if let Some(mirrored_from) = &package_state.mirrored_from {
+                //             kinode_process_lib::print_to_terminal(
+                //                 1,
+                //                 &format!("auto-updating package {package_id} from {mirrored_from}"),
+                //             );
+                //             Request::to(&self.our)
+                //                 .body(serde_json::to_vec(&LocalRequest::Download(
+                //                     DownloadRequest {
+                //                         package_id: crate::kinode::process::main::PackageId::from_process_lib(
+                //                             package_id,
+                //                         ),
+                //                         download_from: mirrored_from.clone(),
+                //                         mirror: package_state.mirroring,
+                //                         auto_update: package_state.auto_update,
+                //                         desired_version_hash: None,
+                //                     },
+                //                 )).unwrap())
+                //                 .send().unwrap();
+                //         }
+                //     }
+                // }
```

Comment:
Should this be uncommented, be removed, or stay as-is?

Reply:
actually that part's still a WIP